tests/daily_scraper/test_invalid_branch.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import mozlog
+import sys
+import unittest
+import urllib
+
+from mozdownload import DailyScraper, NotFoundError
+from mozdownload.utils import urljoin
+
+import mozhttpd_base_test as mhttpd
+
+# testing with an invalid branch parameter should raise an error
+tests_with_invalid_branch = [
+    # -a firefox -t daily -p win32 --branch=invalid
+    {'args': {'application': 'firefox',
+              'branch': 'invalid',
+              'platform': 'win32'}
+     },
+    # -a firefox -t daily -p win32 --branch=invalid --retry-attempts=2
+    {'args': {'application': 'firefox',
+              'branch': 'invalid',
+              'platform': 'win32',
+              'retry_attempts': 2}
+     },
+    # -a firefox -t daily -p win32 --branch=invalid --retry-attempts=2 --retry-delay=0
+    {'args': {'application': 'firefox',
+              'branch': 'invalid',
+              'platform': 'win32',
+              'retry_attempts': 2,
+              'retry_delay': 0.}
```

Comment:
When I ran the tests manually to see the result, I had the same thoughts, but went along with it anyways, because we had it in the list.

If you could let me know what the significance of the test with `retry_delay = 0.`, I would be grateful :)